ctapipe_io_magic/__init__.py

```diff
@@ -961,364 +950,297 @@ def _get_badrmspixel_mask(self, data):
+                if self.is_simulation:
 
-                    data.trigger.tels_with_trigger = np.intersect1d(
-                        data.trigger.tels_with_trigger,
-                        self.subarray.tel_ids,
-                        assume_unique=True
-                    )
+                    event.simulation = SimulatedEventContainer()
 
-                if not self.is_simulation:
+                    event.simulation.shower.energy = event_data['mc_energy'][i_event].to(u.TeV)
+                    event.simulation.shower.shower_primary_id = 1 - event_data['mc_shower_primary_id'][i_event]
+                    event.simulation.shower.h_first_int = event_data['mc_h_first_int'][i_event].to(u.m)
 
-                    data.trigger.tel[tel_id] = TelescopeTriggerContainer(
-                        time=Time(
-                            event_data['unix'][event_i],
-                            format='unix',
-                            scale='utc'
-                        )
-                    )
-
-                    if not generate_pedestals:
-                        badrmspixels_mask = self._get_badrmspixel_mask(data)
-                        data.mon.tel[tel_id].pixel_status.pedestal_failing_pixels = badrmspixels_mask
-
-                # Event counter
-                data.count = counter
-                data.index.obs_id = obs_id
-                data.index.event_id = event_id
-
-                # Setting up the R0 container
-                data.r0.tel.clear()
-                data.r1.tel.clear()
-                data.dl0.tel.clear()
-                data.dl1.tel.clear()
-                data.pointing.tel.clear()
-
-                # Creating the telescope pointing container
-                pointing = PointingContainer()
-                pointing_tel = TelescopePointingContainer(
-                    azimuth=np.deg2rad(event_data['pointing_az'][event_i]) * u.rad,
-                    altitude=np.deg2rad(90 - event_data['pointing_zd'][event_i]) * u.rad,
-                )
+                    # Convert the corsika coordinate (x-axis: magnetic north) to the geographical one.
+                    # Rotate the corsika coordinate by the magnetic declination (= 7 deg):
 
-                pointing.tel[tel_id] = pointing_tel
+                    mfield_dec = self.simulation_config[self.obs_ids[0]]['prod_site_B_declination']
 
-                pointing.array_azimuth = np.deg2rad(event_data['pointing_az'][event_i])*u.rad
-                pointing.array_altitude = np.deg2rad(90 - event_data['pointing_zd'][event_i])*u.rad
-                pointing.array_ra = np.deg2rad(event_data['pointing_ra'][event_i])*u.rad
-                pointing.array_dec = np.deg2rad(event_data['pointing_dec'][event_i])*u.rad
+                    event.simulation.shower.alt = u.Quantity(90, u.deg) - event_data['mc_theta'][i_event].to(u.deg)
+                    event.simulation.shower.az = u.Quantity(180, u.deg) - event_data['mc_phi'][i_event].to(u.deg) + mfield_dec
```

Comment:
While the transformation from Corsika to drive frame is fine, can you please re-add the wrapping up to the [-180,180] interval:

```
if true_az > 180:
    true_az -= 360
```

Reply:
OK sorry for forgetting this, done.